backend/infrahub/webhook/models.py

```diff
@@ -83,28 +180,28 @@ class TransformWebhook(Webhook):
     transform_class: str = Field(...)
     transform_file: str = Field(...)
 
-    async def _prepare_payload(self) -> None:
+    async def _prepare_payload(self, data: dict[str, Any], context: EventContext, service: InfrahubServices) -> None:
         repo: Union[InfrahubReadOnlyRepository, InfrahubRepository]
         if self.repository_kind == InfrahubKind.READONLYREPOSITORY:
             repo = await InfrahubReadOnlyRepository.init(
                 id=self.repository_id,
                 name=self.repository_name,
-                client=self.service.client,
-                service=self.service,
+                client=service.client,
+                service=service,
             )
         else:
             repo = await InfrahubRepository.init(
                 id=self.repository_id,
                 name=self.repository_name,
-                client=self.service.client,
-                service=self.service,
+                client=service.client,
+                service=service,
             )
 
         default_branch = repo.default_branch
         commit = repo.get_commit_value(branch_name=default_branch)
 
         timeout = DEFAULT_TRANSFORM_TIMEOUT
-        if transform := await self.service.client.get(
+        if transform := await service.client.get(
```

Comment:
This doesn't have anything to do with your PR but I notice that we assume that we will get the repo and transform from the default branch here. I don't particularly like any of the options here. Ideally we should target the branch of the specific event if available (depending on the type of event). But it could also be that we've configured webhooks on branches that don't yet have the transform in the branch depending on when it was branched from `main`.